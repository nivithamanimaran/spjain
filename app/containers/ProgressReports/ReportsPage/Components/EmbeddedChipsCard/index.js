import React from 'react';
import {
  Grid,
  Box,
  useTheme,
  Chip,
  Typography,
  makeStyles,
} from '@material-ui/core';
import clsx from 'clsx';

const useStyles = makeStyles(theme => ({
  chip: {
    marginRight: '5px',
    marginBottom: '5px',
    cursor: 'pointer',
    backgroundColor: theme.palette.KenColors.neutral40,
    ' &:hover': {
      backgroundColor: `${theme.palette.KenColors.neutral20}`,
    },
    ' &:focus': {
      backgroundColor: `${theme.palette.KenColors.kenWhite}`,
      border: `1px solid ${theme.palette.KenColors.tertiaryBlue500}`,
    },
  },
  variantGoodChip: {
    backgroundColor: `${theme.palette.KenColors.kenWhite}`,
    ' &:focus': {
      backgroundColor: `${theme.palette.KenColors.kenWhite}`,
      border: `1px solid ${theme.palette.KenColors.tertiaryGreen500}`,
    },
  },
  variantBadChip: {
    backgroundColor: `${theme.palette.KenColors.kenWhite}`,
    ' &:focus': {
      backgroundColor: `${theme.palette.KenColors.kenWhite}`,
      border: `1px solid ${theme.palette.KenColors.tertiaryRed500}`,
    },
  },
}));

const EmbeddedChipsCard = props => {
  const theme = useTheme();
  const classes = useStyles();
  const {
    title,
    cardBackgroundColor,
    chipData,
    keyInArray,
    chipProps,
    cardTitleColor,
    chipTextColor,
    variant,
    onChipClick,
    onChipDelete,
    focusedChipStyles, //make sure to pass styles which are defined at right place.
  } = props;

  let focusedChipClass = {};
  if (focusedChipStyles) {
    focusedChipClass = focusedChipStyles;
  }
  const getCardTitleColor = () => {
    if (cardTitleColor) return cardTitleColor;

    switch (variant) {
      case 'good':
        return theme.palette.KenColors.tertiaryGreen500;
      case 'bad':
        return theme.palette.KenColors.tertiaryRed500;

      default:
        return theme.palette.KenColors.primary;
    }
  };

  const getCardBgColor = () => {
    if (cardBackgroundColor) return cardBackgroundColor;

    switch (variant) {
      case 'good':
        return theme.palette.KenColors.tertiaryGreen501;
      case 'bad':
        return theme.palette.KenColors.secondaryPeach2;

      default:
        return theme.palette.KenColors.kenWhite;
    }
  };

  const getChipTextColor = () => {
    if (chipTextColor) return chipTextColor;

    switch (variant) {
      case 'good':
        return theme.palette.KenColors.tertiaryGreen502;
      case 'bad':
        return theme.palette.KenColors.secondaryPeach4;

      default:
        return theme.palette.KenColors.primary;
    }
  };

  const getChipLabel = item => {
    if (typeof item === 'object') {
      return item[keyInArray] ? item[keyInArray] : item['label'];
    } else if (typeof item === 'string') {
      return item;
    }
  };

  return (
    <Box
      style={{
        borderRadius: '10px',
        backgroundColor: getCardBgColor(),
      }}
      data-testid="embedded-chips-card"
    >
      <Box>
        <Box p={2}>
          <Typography
            data-testid="chips-card-title"
            style={{
              fontWeight: 'bold',
              color: getCardTitleColor(),
            }}
            variant="subtitle2"
          >
            {title}
          </Typography>
        </Box>
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <Box p={1}>
              {Array.isArray(chipData) &&
                chipData.map((item, index) => {
                  return (
                    <>
                      {(item[keyInArray] ||
                        item['label'] ||
                        typeof item === 'string') && (
                        <Chip
                          key={index}
                          data-testid="chips-label"
                          className={clsx(classes.chip, focusedChipClass, {
                            [classes.variantGoodChip]: variant === 'good',
                            [classes.variantBadChip]: variant === 'bad',
                          })}
                          clickable={true}
                          onClick={e => {
                            onChipClick(item, e);
                          }}
                          onDelete={
                            onChipDelete
                              ? e => {
                                  onChipDelete(item, e);
                                }
                              : null
                          }
                          label={
                            <Typography
                              variant="subtitle2"
                              style={{
                                color: getChipTextColor(),
                                fontWeight: 'bold',
                              }}
                            >
                              {getChipLabel(item)}
                            </Typography>
                          }
                          {...chipProps}
                        />
                      )}
                    </>
                  );
                })}
            </Box>
          </Grid>
        </Grid>
      </Box>
    </Box>
  );
};
export default EmbeddedChipsCard;
