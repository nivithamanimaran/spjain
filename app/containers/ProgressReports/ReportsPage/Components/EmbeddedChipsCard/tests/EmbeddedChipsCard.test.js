import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import EmbeddedChipsCard from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Embedded-Chips-Card', () => {
  const div = document.createElement('div');
  it('Embedded Chips Card renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <EmbeddedChipsCard />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <EmbeddedChipsCard />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('embedded-chips-card')).toMatchSnapshot();
  });
  it('Ratings status Card renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <EmbeddedChipsCard boldText="bold" boldSubtext="subtext" />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <EmbeddedChipsCard
            title="Title"
            chipData={[{ label: 'Hindi', value: 'hindi' }]}
          />
        </Router>
      </ThemeProvider>
    );

    const title = queryByTestId('chips-card-title');
    expect(title).toBeTruthy();
    expect(title.textContent).toBe('Title');

    const label = queryByTestId('chips-label');
    expect(label).toBeTruthy();
    expect(label.textContent).toBe('Hindi');
  });
});
