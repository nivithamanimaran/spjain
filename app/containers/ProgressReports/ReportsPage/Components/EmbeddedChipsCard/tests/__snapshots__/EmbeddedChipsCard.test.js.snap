// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embedded-Chips-Card Embedded Chips Card renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-7"
  data-testid="embedded-chips-card"
  style="border-radius: 10px; background-color: rgb(255, 255, 255);"
>
  <div
    class="MuiBox-root MuiBox-root-8"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2"
      style="font-weight: bold; color: rgb(0, 33, 141);"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-9"
      />
    </div>
  </div>
</div>
`;

exports[`Embedded-Chips-Card Embedded Chips Card renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-7"
  data-testid="embedded-chips-card"
  style="border-radius: 10px; background-color: rgb(255, 255, 255);"
>
  <div
    class="MuiBox-root MuiBox-root-8"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2"
      data-testid="chips-card-title"
      style="font-weight: bold; color: rgb(0, 33, 141);"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-9"
      />
    </div>
  </div>
</div>
`;
